test/e2e/cro_test.go

```diff
@@ -0,0 +1,190 @@
+	timeout       = time.Second * 160
+)
+
+func TestCRO(t *testing.T) {
+	// adding redis scheme to framework
+	redisList := &v1alpha1.Redis{}
+	if err := framework.AddToFrameworkScheme(apis.AddToScheme, redisList); err != nil {
+		t.Fatalf("failed to add Redis custom resource scheme to framework: %v", err)
+	}
+
+	// adding postgres scheme to framework
+	postgresList := &v1alpha1.Postgres{}
+	if err := framework.AddToFrameworkScheme(apis.AddToScheme, postgresList); err != nil {
+		t.Fatalf("failed to add Postgres custom resource scheme to framework: %v", err)
+	}
+
+	// run subtests
+	t.Run("cro-group", func(t *testing.T) {
+		t.Run("Cluster", CROCluster)
+	})
+}
+
+func CROCluster(t *testing.T) {
+	t.Parallel()
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+	err := ctx.InitializeClusterResources(getCleanupOptions(t))
+	if err != nil {
+		t.Fatalf("failed to initialize cluster resources: %v", err)
+	}
+	t.Log("initialized cluster resources")
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+	// get global framework variables
+	f := framework.Global
+	// wait for cloud-resource-operator to be ready
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "cloud-resource-operator", 1, retryInterval, timeout)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// run postgres test
+	if err = postgresBasicTest(t, f, *ctx); err != nil {
+		t.Fatal(err)
+	}
+}
+
+// basic test, creates postgres resource, checks deployment has been created, the status has been updated.
+// the secret has been created and populated, deletes the postgres resource and checks all resources has been deleted
+func postgresBasicTest(t *testing.T, f *framework.Framework, ctx framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return errorUtil.Wrapf(err, "could not get namespace")
+	}
+	examplePostgres := &v1alpha1.Postgres{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      postgresName,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.PostgresSpec{
+			SecretRef: &v1alpha1.SecretRef{
+				Name:      "example-postgres-sec",
+				Namespace: namespace,
+			},
+			Tier: "development",
+			Type: "workshop",
+		},
+	}
+	// create postgres resource
+	if err := f.Client.Create(goctx.TODO(), examplePostgres, getCleanupOptions(t)); err != nil {
+		return errorUtil.Wrapf(err, "could not create example Postgres")
+	}
+	t.Logf("created %s resource", examplePostgres.Name)
+
+	// wait from postgres deployment
+	if err := e2eutil.WaitForDeployment(t, f.KubeClient, namespace, postgresName, 1, retryInterval, timeout); err != nil {
+		return errorUtil.Wrapf(err, "could not get deployment")
+	}
+	t.Logf("%s deployment created", postgresName)
+
+	// poll cr for complete status phase
+	pcr := &v1alpha1.Postgres{}
+	err = wait.Poll(retryInterval, time.Minute*6, func() (done bool, err error) {
+		if err := f.Client.Get(goctx.TODO(), types.NamespacedName{Namespace: namespace, Name: postgresName}, pcr); err != nil {
```

Comment:
could we add a log statement in here letting the user know we're polling for complete and the current status phase is, just so we know what's happening when looking at this from a ci perspective?

Reply:
I had it there, but the logs, are only printed after the test is run, so it added noise to the final output. 
Ill have a look to see if i can get them printed out in real time, and update.

Reply:
ah that's sucky, but good to know for people when they're writing tests.

it makes sense to keep it clean then.